### ISSUE TITLE
Application window size reset to 800px x 630px after restart if ShowTrayIcon=true

### ISSUE THREAD

**Issue by thxpian** [keepassxreboot/keepassxc#9734]
## Steps to Reproduce
1. Enable KPXC setting “Minimize window at application startup“ & “Show tray icon“.
2. Quit KPXC and restart.

## Expected Behavior
- The window size that the user has chosen should NOT be altered / reset.

## Actual Behavior
- The application window is resized to width:800px, height: ~630px, ignoring whatever I resized it to before.

KeePassXC - 2.7.5
Revision: 9d0537b

Operating System: tested on macOSX 10.15.7 and macOS 13.5

**Comment by droidmonkey** [comment]
Close KeePassXC, then go to: `/Users/[your username]/Library/Caches/KeePassXC` and delete keepassxc.ini 

You might have to show hidden files, press `CMD + Shift + .` in Finder to do that.

Now load KeePassXC and setup your desired settings.

You might also achieve success by going to application settings within KeePassXC, scroll to the bottom, then hit "Reset to Defaults", but that'll wipe out all your other settings as well.

**Comment by thxpian** [comment]
Wiping the settings does not have the desired effect.

I need to amend my Steps To Reproduce:
It happens if you set
```
ShowTrayIcon=true
MinimizeOnStartup=true
```
The **ShowTrayIcon** makes the difference, it seems.

**Comment by droidmonkey** [comment]
What do you mean by "compact mode"?

**Comment by thxpian** [comment]
I mistakenly asssociated this with _View > Compact Mode_. Probably because changing that required a restart. I've edited the issue title and my first post accordingly.

If you (re)start KPXC using the settings with `ShowTrayIcon` mentioned above, the application window is always resized to width:800px, height: ~630px, ignoring whatever I resized it to before.

**Comment by droidmonkey** [comment]
Those settings have nothing to do with each other. Show tray icon just shows a tray icon. It doesn't adjust the window size in any way. 

If your window size settings are being forgotten, then that has to do with potentially not keeping the cache configuration file. The one I pointed to above.

**Comment by thxpian** [comment]
Look, I'm not trying to be deliberately obstinate here. And I'm not postulating any specific causality.
All I can do is report what happens here with my everyday install of KPXC  on macOS 13.5 and with a fresh test install on MacOSX 10.15.7. I wouldn't know what's causing this exactly. All I know is that toggling `ShowTrayIcon` makes a difference, at least here on my OS. 
But I assume you have a lot on your plate and I understand if you don't want to address this any futher.

One last thing I can do is post excerpts from the respective _keepassxc.ini_ files: 
### **No** resizing
````
[General]
ConfigVersion=2
MinimizeAfterUnlock=true
UpdateCheckMessageShown=true

[GUI]
CheckForUpdates=false
ColorPasswords=true
MinimizeOnClose=true
MinimizeOnStartup=true
MinimizeToTray=true
ShowTrayIcon=false
TrayIconAppearance=monochrome
````

### Unwanted resizing (after app restart)
````
[General]
ConfigVersion=2
MinimizeAfterUnlock=true
UpdateCheckMessageShown=true

[GUI]
CheckForUpdates=false
ColorPasswords=true
MinimizeOnClose=true
MinimizeOnStartup=true
MinimizeToTray=true
ShowTrayIcon=true
TrayIconAppearance=monochrome
````

Since you haven't been able to repoduce this, I won't trouble you any further.
Thank you, nonetheless, for this application. Greatly appreciated.

**Comment by droidmonkey** [comment]
Ahhh you might be running into the crash on close bug which might be causing your config settings to not be saved (specifically the window size). That is fixed in upcoming 2.7.6

**Comment by rzelazko** [comment]
The issue persists on the 2.7.10 version. Thanks to @thxpian, we have a workaround: change `MinimizeOnStartup` to false, then the window size will be persisted on the next run. 
If minimize on startup is turned on, the window will be back to its default size.
